services/cd-service/pkg/service/overview.go

```diff
@@ -83,8 +83,20 @@ func (o *OverviewServiceServer) getOverviewDB(
 	if s.DBHandler.ShouldUseOtherTables() {
 		err := s.DBHandler.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
 			var err2 error
+			cached_result, err2 := s.DBHandler.ReadLatestOverviewCache(ctx, transaction)
+			if err2 != nil {
+				return err2
+			}
+			if !s.DBHandler.CheckOverviewIsEmpty(cached_result) {
+				response = cached_result
```

Comment:
I think this function would be more easy to follow, if we used `WithTransactionT` here and then `return cached_result`.